### DISCUSSION TITLE
Script to empty all logs in all containers.

### DISCUSSION THREAD

**Discussion by larod241** [tteck/Proxmox#991] [Ideas]
Hello,

would it be possible to make a script that would automatically purge the logs of all containers?
/var/log/syslog
/var/log/syslog.*
/var/log/demon.log
/var/log/demon.*
etc etc

Thanks !

**Comment by tteck** [comment]
How about `rm -rf /var/{cache,log}/* /var/lib/apt/lists/*` for a good clean

**Reply by larod241** [reply]
Thanks for the command lines. 

Would it be possible to integrate it in a script in order to process all containers at once?

**Reply by tteck** [reply]
I think a "Proxmox LXC Cleaner" can be arranged.

**Reply by larod241** [reply]
you are a genius!!! 
Thank you again for all these scripts that simplify, to all of us, the life!

**Reply by tteck** [reply]
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/clean-lxcs.sh)"
```
Let me know watcha think

**Reply by tteck** [reply]
only for apt package manager (Debian/Ubuntu)

**Reply by tteck** [reply]
**_UPDATED_**
What the script does:
- `apt-get -y --purge autoremove`
- `apt-get -y autoclean` 
- `find /var/cache -type f` (Lists cache then gives option to delete)
- `find /var/log -type f` (Lists logs then gives option to delete)
- `rm -rf  /var/lib/apt/lists/*`  (removes apt lists)
- `apt-get update` (repopulate apt lists)

Leaves directory structure intact

**Reply by larod241** [reply]
> ```
> bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/clean-lxcs.sh)"
> ```
> 
> 
>      
>   
> 
> Let me know watcha think

I've juste test it and it work fine !!! I've check 5 lxc of my 15+ lxc and log are gone ! 

Thank you !!!

**Reply by smtdev** [reply]
Amazing, thank you @tteck.
